### PR TITLE
New User

### PR DESCRIPTION
**Pull request by A-nastasiya**

### CODE REVIEW

**Comment by A-nastasiya** [comment]
https://github.com/DmitryLeviossa/leviossa/pull/23

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -6,4 +6,10 @@ def index
   def show
     @user = current_company.users.find_by(id: params[:id])
   end
+
+  def create_user
```
Comment (DmitryLeviossa) [inline app/controllers/users_controller.rb]: name it just create

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -6,4 +6,10 @@ def index
...
+    company = Company.create(name: params[:users][:name])
```
Comment (DmitryLeviossa) [inline app/controllers/users_controller.rb]: don't create new company for that, should be current_company

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -6,4 +6,10 @@ def index
...
+    department = Department.create(name: company.name, company_id: company.id)
```
Comment (DmitryLeviossa) [inline app/controllers/users_controller.rb]: no need to create department, add select field to form which will send department_id, can take samle on projects modal

File: app/javascript/bundles/components/users/UserFormModal.js [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+import {Section, TextInput, FlashErrors, CustomInput, TextAreaInput} from '../../helpers/elementsHelpers';
+import axios from 'axios-on-rails';
+import {Async} from 'react-select';
+
+
+export default class usersFormModal extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            users: this.props.users || {}
```
Comment (DmitryLeviossa) [inline app/javascript/bundles/components/users/UserFormModal.js]: Why it's plural?

File: app/javascript/bundles/components/users/UserFormModal.js [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+                            label='Password'
+                            onChange={this.changeInput.bind(this, 'password')}
+                            value={this.state.users.password}/>
+                        <TextInput
+                            label='Name'
+                            onChange={this.changeInput.bind(this, 'name')}
+                            value={this.state.users.name}/>
+                        <CustomInput label='Role' data-columnNumber="0">
+                            <Async
+                                loadOptions={getOptions}
```
Comment (DmitryLeviossa) [inline app/javascript/bundles/components/users/UserFormModal.js]: Why for roles you use customers?

File: config/routes.rb [hunk 6]
```diff
@@ -11,8 +11,7 @@
   end
   resources :departments, except: [:new, :edit]
   resources :company, only: :index
-  resources :users, only: [:index, :show, :switch] do
-    post :create_user, on: :collection
-  end
+  resources :users, only: [:index, :show, :switch]
+  post 'users/create', to: 'users#create'
```
Comment (DmitryLeviossa) [inline config/routes.rb]: change only to exept and put destroy, :new, :edit and remove create rout

File: app/javascript/bundles/components/users/UserFormModal.js (hunk 5 above)
Comment (DmitryLeviossa) [reply app/javascript/bundles/components/users/UserFormModal.js]: You should has select async for departments

File: app/controllers/users_controller.rb [hunk 7]
```diff
@@ -7,9 +7,13 @@ def show
     @user = current_company.users.find_by(id: params[:id])
   end
 
-  def create_user
-    company = Company.create(name: params[:users][:name])
-    department = Department.create(name: company.name, company_id: company.id)
-    User.create(email: params[:users][:email],password: params[:users][:password], company_id: company.id, department_id: department.id )
+  def create
+    @user = User.new(email: params[:user][:email], password: params[:user][:password], company_id: current_company.id, department_id: current_company.departments.last.id)
```
Comment (DmitryLeviossa) [inline app/controllers/users_controller.rb]: created method in private block that will be called in new()

File: app/javascript/bundles/components/users/UserFormModal.js [hunk 8]
```diff
@@ -5,52 +5,50 @@ import axios from 'axios-on-rails';
 import {Async} from 'react-select';
 
 
-export default class usersFormModal extends React.Component {
+export default class userFormModal extends React.Component {
```
Comment (DmitryLeviossa) [inline app/javascript/bundles/components/users/UserFormModal.js]: all js components should be capitalized

File: app/javascript/bundles/components/users/UserFormModal.js [hunk 9]
```diff
@@ -5,52 +5,50 @@ import axios from 'axios-on-rails';
...
         this.state = {
-            users: this.props.users || {}
+            user: this.props.user || {}
         };
     }
 
-    componentWillReceiveProps(props) {
-        this.setState({users: props.users || {}, errors: []});
-    }
+    // componentWillReceiveProps(props) {
```
Comment (DmitryLeviossa) [inline app/javascript/bundles/components/users/UserFormModal.js]: componentWillReceiveProps is required for edit

File: app/javascript/bundles/components/users/UserFormModal.js [hunk 10]
```diff
@@ -5,52 +5,50 @@ import axios from 'axios-on-rails';
...
+    //     this.setState({user: props.user || {}, errors: []});
+    // }
 
     handleSave() {
-        let url = '/users/create_user',
+        let url = '/users/create',
```
Comment (DmitryLeviossa) [inline app/javascript/bundles/components/users/UserFormModal.js]: it should be just /users with post request

**Pull request merged**